### PR TITLE
Update readme.

### PR DESCRIPTION
**Pull request by miladHakimi**
- Add the Trace Event's environment variable to the suggested environment variables list for the log files.
- Add a new section, log formats, describing supported formats and their use cases.
- Add a screenshot of the output generated by Perfetto.
- Fix the names of some of the environment variables.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ This project contains 5 Vulkan layers:
...
 5. Device memory usage layer. This layer tracks memory explicitly allocated by the application (VkAllocateMemory), usually for images and buffers. For each frame, current allocation and maximum allocation is written to the log file. The output log file location can be set with the `VK_MEMORY_USAGE_LOG` environment variable.
 
-The results are saved as CSV files. Setting the `VK_PERFORMANCE_LAYERS_EVENT_LOG_FILE` environment variable makes all layers append their events (with timestamps) to a single file.
+The results are saved as CSV files. Setting the `VK_PERFORMANCE_LAYERS_EVENT_LOG_FILE` and `VK_PERFORMANCE_LAYERS_TRACE_EVENT_LOG_FILE` environment variables makes all layers append their events (with timestamps) to these file. Both of these files contain all the events but in different formats.
```
Comment (kuhar) [inline README.md]: Could you make this sentence into two bullets, once for each format? I think it will be easier to parse

File: README.md [hunk 2]
```diff
@@ -75,20 +75,21 @@ export LD_LIBRARY_PATH=<path-with-layers.so>:$LD_LIBRARY_PATH
 1. VK_LAYER_STADIA_pipeline_compile_time
 1. VK_LAYER_STADIA_pipeline_runtime
 1. VK_LAYER_STADIA_pipeline_cache_sideload
-1. VK_LAYER_STADIA_pipeline_memory_usage
-1. VK_LAYER_STADIA_pipeline_frame_time
+1. VK_LAYER_STADIA_memory_usage
+1. VK_LAYER_STADIA_frame_time
```
Comment (kuhar) [inline README.md:112]: Thanks!

File: README.md [hunk 3]
```diff
@@ -97,6 +98,17 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the python scripts in the `scripts/` directory and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. see:
```
Comment (kuhar) [inline README.md]: nit: To learn more about the trace event format, see: the-link.

File: README.md [hunk 4]
```diff
@@ -97,6 +98,17 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
...
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+Our logging system also supports custom formats (e.g., `sample_output/events.log`).
+
+#### Adding a new log format:
+
+To add a custom format, create a new logger by extending `EventLogger` class, the base class for all the loggers. Then, introduce the new logger to the `LayerData` through its constructor.
```
Comment (kuhar) [inline README.md]: This fits developer documentation more than the user-facing one. The readme is primarily aimed at end users. I'd leave this out.

File: README.md [hunk 5]
```diff
@@ -97,6 +98,17 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the python scripts in the `scripts/` directory and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. see:
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+Our logging system also supports custom formats (e.g., `sample_output/events.log`).
```
Comment (kuhar) [inline README.md]: You can link to the file: https://github.blog/2013-01-31-relative-links-in-markup-files/

File: README.md [hunk 6]
```diff
@@ -97,6 +98,17 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the python scripts in the `scripts/` directory and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. see:
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
```
Comment (kuhar) [inline README.md]: Could you add a screenshot?

**Review by kuhar — CHANGES_REQUESTED** [review]
Could you add a few bullet with the summary of the changes to the PR description?

File: README.md (hunk 6 above)
Comment (miladHakimi) [reply README.md]: Sure.

File: README.md [hunk 7]
```diff
@@ -7,7 +7,9 @@ This project contains 5 Vulkan layers:
...
 5. Device memory usage layer. This layer tracks memory explicitly allocated by the application (VkAllocateMemory), usually for images and buffers. For each frame, current allocation and maximum allocation is written to the log file. The output log file location can be set with the `VK_MEMORY_USAGE_LOG` environment variable.
 
-The results are saved as CSV files. Setting the `VK_PERFORMANCE_LAYERS_EVENT_LOG_FILE` environment variable makes all layers append their events (with timestamps) to a single file.
+The results are saved as CSV files. Setting the following environment variables makes all layers append their events (with timestamps) to these file, both contain all the events but in different formats.
```
Comment (kuhar) [inline README.md]: nit: these files

We should further break this down into these two bullets IMO. The new formats are not CSV files like the first sentence suggests. What do you think about starting a new paragraph and describing these two formats in separation?

File: README.md [hunk 8]
```diff
@@ -97,6 +100,18 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
...
+
+To visualize the logs, we use multiple trace viewers such as the analysis scripts in the [scripts directory](./scripts/) and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. To learn more about the Trace Event format, see:
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+Here is a sample output generated by Perfetto:
+    ![Timeline View](sample_output/perfetto.png)
+
+
+Our logging system also supports custom formats (e.g., [events.log](./sample_output/events.log)
+).
```
Comment (kuhar) [inline README.md]: nit: move this to the line above?

File: README.md [hunk 9]
```diff
@@ -97,6 +100,18 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the analysis scripts in the [scripts directory](./scripts/) and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. To learn more about the Trace Event format, see:
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+Here is a sample output generated by Perfetto:
```
Comment (kuhar) [inline README.md]: nit: Sample output as rendered by Perfetto:

**Review by kuhar — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -97,6 +100,18 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the analysis scripts in the [scripts directory](./scripts/) and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. To learn more about the Trace Event format, see:
+https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
```
Comment (kuhar) [inline README.md]: nit: missing `.` at the end of the sentence.

File: README.md (hunk 7 above)
Comment (miladHakimi) [reply README.md]: Sounds good!

File: README.md [hunk 11]
```diff
@@ -97,6 +100,18 @@ export VK_COMPILE_TIME_LOG=<log-file-path>
 
 To check if the layers are enabled, you can run a sample Vulkan application (such as `vkcube`) and look for the generated layer log files, or check the [Vulkan loader logs](https://github.com/KhronosGroup/Vulkan-Loader/blob/master/docs/LoaderInterfaceArchitecture.md#table-of-debug-environment-variables).
 
+### Log formats:
+
+To visualize the logs, we use multiple trace viewers such as the analysis scripts in the [scripts directory](./scripts/) and Perfetto (https://ui.perfetto.dev/). To suit the input format requirements of each logger, we support logging in various formats such as CSV and Chrome Trace Event. To learn more about the Trace Event format, see:
```
Comment (kuhar) [inline README.md]: I'm not a fan of this description:
1. Doesn't get to the point quickly enough.
2. Format requirements are irrelevant to end users and how we suit them doesn't matter.
3. Common logs are not CSV files.

Instead, should we list all supported formats, create a paragraph describing each format, what it is useful for, what are the properties, which flag enables it, etc.?

**Review by kuhar — APPROVED** [review]
Neat, thanks for the changes!

**Pull request merged**